src/components/Dao/DaoWiki.tsx

```diff
@@ -0,0 +1,167 @@
+import { WikiContainer, actualHash, ReactiveWiki } from "@dorgtech/daosmind";
+import classNames from "classnames";
+import { enableWalletProvider } from "arc";
+import { combineLatest } from "rxjs";
+
+import { Link } from "react-router-dom";
+import * as arcActions from "actions/arcActions";
+import { showNotification } from "reducers/notifications";
+import { schemeName } from "lib/util";
+import { connect } from "react-redux";
+import { RouteComponentProps } from "react-router-dom";
+import withSubscription, { ISubscriptionProps } from "components/Shared/withSubscription";
+import Loading from "components/Shared/Loading";
+import * as proposalStyle from "../Scheme/SchemeProposals.scss";
+import * as daoStyle from "./Dao.scss";
+
+type IExternalProps = {
+  daoState: IDAOState;
+  match: Record<string, any>;
+} & RouteComponentProps<any>;
+
+const mapDispatchToProps = {
+  createProposal: arcActions.createProposal,
+  showNotification,
+};
+
+interface IDispatchProps {
+  createProposal: typeof arcActions.createProposal;
+  showNotification: typeof showNotification;
+}
+
+type SubscriptionData = ISubscriptionProps<[Scheme[], Proposal[]]>
+type IProps = IDispatchProps & IExternalProps & SubscriptionData;
+
+function DaoWiki(props: IProps) {
+  const [hasWikiScheme, setHasWikiScheme] = React.useState<boolean>(false);
+  const [schemes, proposals] = props.data;
+
+  const renderWikiComponent = () => {
+    const { daoAvatarAddress, perspectiveId, pageId } = props.match.params;
+    actualHash["dao"] = daoAvatarAddress;
+    actualHash["wiki"] = perspectiveId;
+    actualHash["page"] = pageId;
+    return WikiContainer.getInstance({});
+  };
+
+  const checkIfWikiSchemeExists = async () => {
+    const genericSchemes = schemes.filter((scheme: Scheme) => scheme.staticState.name === "GenericScheme")
+    const states: ISchemeState[] = [];
+    const getSchemeState = () => {
+      return new Promise((resolve, reject) => {
+        try {
+          genericSchemes.map((scheme: Scheme) => scheme.state().subscribe(state => states.push(state)));
+          resolve();
+        } catch (e) {
+          reject(e);
+        }
+      });
+    };
+    await getSchemeState();
+    const checkSchemes = (schemeState: ISchemeState) => {
+      return "WikiUpdate" === schemeName(schemeState, "[Unknown]");
+    };
+    const wikiSchemeExists = states.some(checkSchemes);
+    setHasWikiScheme(wikiSchemeExists);
+    if (wikiSchemeExists) {
+      renderWikiComponent();
+    }
+  };
+
+  React.useEffect(() => {
+    checkIfWikiSchemeExists();
+    console.log(proposals)
+  }, []);
+
+  const registerWikiScheme = async () => {
+    if (!await enableWalletProvider({ showNotification: props.showNotification })) { return; }
+    
+    const registrarScheme = schemes.filter((scheme: Scheme) => scheme.staticState.name === "SchemeRegistrar").pop();
+    const registrarSchemeState = () => {
```

Comment:
don't you already have the scheme states from when you fetch dao.schemes and set fetchAllData to true?

Reply:
Same here, we need the ID and Address of the `registrarScheme`, the staticState was only retrieving the ID.

As said before, if you think this can/should be improved just let us know.

Reply:
I just checked this one, and you're totally right. There's not need of calling the `.state` method here, fixing it. Thanks!